python/cdp/to_network_scoped_evm_server_account.py

```diff
@@ -0,0 +1,331 @@
+"""Functions to convert existing EVM server accounts to network-scoped versions."""
+
+from collections.abc import Callable
+from typing import Any, Literal
+
+from eth_account import Account as EthAccount
+from web3 import Web3
+from web3.middleware import geth_poa_middleware
+
+from cdp.actions.evm.swap import AccountSwapOptions
+from cdp.evm_server_account import EvmServerAccount
+from cdp.network_capabilities import is_method_supported_on_network
+
+
+class ToNetworkScopedEvmServerAccountOptions:
+    """Options for converting a pre-existing EvmAccount to a NetworkScopedEvmServerAccount."""
+
+    def __init__(self, account: EvmServerAccount, network: str):
+        """Initialize the options.
+
+        Args:
+            account: The EvmServerAccount that was previously created.
+            network: The network to scope the account to.
+
+        """
+        self.account = account
+        self.network = network
+
+
+class NetworkScopedEvmServerAccount:
+    """A network-scoped EVM server account that only exposes methods supported by the network.
+
+    Uses dynamic attribute access to match the TypeScript approach.
+    """
+
+    def __init__(
+        self,
+        evm_server_account: EvmServerAccount,
+        network: str,
+        rpc_url: str | None = None,
+        private_key: str | None = None,
```

Comment:
Should use the EvmServerAccount that is instantiating the `NetworkScopedEvmServerAccount`